pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/performance/UseStringBufferForStringAppendsRule.java

```diff
@@ -39,92 +38,65 @@ public UseStringBufferForStringAppendsRule() {
     @Override
     public Object visit(ASTVariableDeclaratorId node, Object data) {
         if (!TypeTestUtil.isA(String.class, node) || node.hasArrayType()
-                || node.getNthParent(3) instanceof ASTForStatement) {
+                || node.getNthParent(3) instanceof ASTForStatement
+                || node.getNthParent(3) instanceof ASTForeachStatement) {
```

Comment:
Magic number: this looks like we could use a method `isForeachVariable`

Reply:
Also we shouldn't be checking for ForStatement. The original check only catches foreaches, because in a for loop, it's the 4th parent which is the ASTForStatement (there's the ASTForInit in between). This makes sense, because the variable of a foreach is local to each iteration, while for a for loop, it's the same in all iterations.